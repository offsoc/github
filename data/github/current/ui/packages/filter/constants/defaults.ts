import {
  type FilterConfig,
  type FilterProviderOptions,
  type FilterSettings,
  NOT_SHOWN,
  ProviderSupportStatus,
} from '../types'

export const defaultFilterConfig: FilterConfig = {
  filterDelimiter: ':',
  valueDelimiter: ',',
  variant: 'full',
}

export const defaultFilterProviderOptions: FilterProviderOptions = {
  priority: NOT_SHOWN,
  filterTypes: {
    inclusive: true,
    exclusive: true,
    valueless: true,
    multiKey: true,
    multiValue: true,
  },
  support: {
    status: ProviderSupportStatus.Supported,
  },
}

export const defaultFilterSettings: FilterSettings = {
  aliasMatching: false,
  disableAdvancedTextFilter: false,
}
