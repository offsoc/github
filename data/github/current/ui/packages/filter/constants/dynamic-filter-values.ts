import {ArrowSwitchIcon, CalendarIcon} from '@primer/octicons-react'

import type {FilterSuggestion} from '../types'

function getCurrentDate() {
  return new Date().toLocaleDateString('en-us', {year: 'numeric', month: 'long', day: 'numeric'})
}

export const TIME_RANGE_VALUES: FilterSuggestion[] = [
  {
    value: '@today',
    displayName: 'Today',
    priority: 1,
    icon: ArrowSwitchIcon,
    iconColor: 'var(--fgColor-done, var(--color-done-fg))',
  },
  {
    value: '@today-1d',
    displayName: 'Yesterday',
    priority: 2,
    icon: ArrowSwitchIcon,
    iconColor: 'var(--fgColor-done, var(--color-done-fg))',
  },
  {
    value: '>@today-1w',
    displayName: 'Past 7 days',
    priority: 3,
    icon: ArrowSwitchIcon,
    iconColor: 'var(--fgColor-done, var(--color-done-fg))',
  },
  {
    value: '>@today-30d',
    displayName: 'Past 30 days',
    priority: 4,
    icon: ArrowSwitchIcon,
    iconColor: 'var(--fgColor-done, var(--color-done-fg))',
  },
  {
    value: '>@today-1y',
    displayName: 'Past year',
    priority: 5,
    icon: ArrowSwitchIcon,
    iconColor: 'var(--fgColor-done, var(--color-done-fg))',
  },
  {
    value: () => {
      const now = new Date()
      return `${now.getFullYear()}-${`0${now.getMonth() + 1}`.slice(-2)}-${`0${now.getDate()}`.slice(-2)}`
    },
    displayName: getCurrentDate(),
    priority: 6,
    icon: CalendarIcon,
  },
]

export const NUMBER_COUNT_VALUE = [
  {value: '<10', displayName: 'Less than 10', priority: 1},
  {value: '>10', displayName: 'More than 10', priority: 2},
  {value: '10..100', displayName: 'Between 10 and 100', priority: 3},
  {value: '100', displayName: '100', priority: 4},
]
