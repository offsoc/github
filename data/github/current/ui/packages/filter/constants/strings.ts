import {MAX_NESTED_GROUPS} from './filter-constants'

export const Strings = {
  filterInvalid: (number: number) => `Filter contains ${number} issue${number === 1 ? '' : 's'}:`,
  filterBlockInvalid: (block: string) => `${block} is invalid`,
  filterProviderNotSupported: (filterName: string) => `Filter '${filterName}' not supported`,
  filterProviderDeprecated: (filterName: string) =>
    `Filter '${filterName}' is deprecated and will be removed in the future`,
  filterValueEmpty: (provider: string) => `Empty value for <pre>${provider}</pre>`,
  filterMultiValueFalse: (provider: string) => `Only one value allowed for <pre>${provider}</pre>`,
  filterInvalidValue: (provider: string, value: string) =>
    `Invalid value <pre>${value}</pre> for <pre>${provider}</pre>`,
  unbalancedQuotations: 'Unbalanced quotation marks',
  unbalancedParentheses: 'Unbalanced parentheses',
  maxNestedGroups: `Maximum nested groups of ${MAX_NESTED_GROUPS} exceeded`,
  advancedFilterDialogCloseConfirmation: {
    title: 'Discard changes?',
    content: 'You have unsaved changes. Are you sure you want to discard them?',
    cancelButtonContent: 'Keep editing',
    confirmButtonContent: 'Close and discard',
  },
  exclude: 'Exclude',
  dividerValue: '%%DIVIDER%%',
}
