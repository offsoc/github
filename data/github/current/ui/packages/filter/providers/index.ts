export {AsyncFilterProvider} from './async'
export * from './key-only'
export {LabelFilterProvider} from './label'
export {LanguageFilterProvider} from './language'
export {MilestoneFilterProvider} from './milestone'
export {OrgFilterProvider} from './org'
export {ProjectFilterProvider} from './project'
export {RepositoryFilterProvider} from './repository'
export * from './static'
export {TeamFilterProvider, TeamReviewRequestedFilterProvider} from './team'
export {TypeFilterProvider} from './type'
export * from './user'
