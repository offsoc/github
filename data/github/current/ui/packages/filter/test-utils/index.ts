import {act, fireEvent, screen} from '@testing-library/react'

export const setupExpectedAsyncErrorHandler = () => {
  jest.spyOn(console, 'error').mockImplementation((message: string) => {
    // * Because Filter is asynchronous, there are console errors that are thrown, but expected. This will rethrow
    // * any errors that are not related to the async nature of the component.
    if (!message.includes?.('wrapped in act(')) {
      // eslint-disable-next-line no-console
      console.warn(message)
    }
  })
}

export async function updateFilterValue(key: string, value: string = '') {
  jest.useFakeTimers()

  const input = screen.getByRole('combobox')
  input.focus()

  fireEvent.change(input, {target: {value: `${key}${key && value ? ':' : ''}${value}`}})

  // eslint-disable-next-line @typescript-eslint/require-await
  await act(async () => {
    jest.runAllTimers()
  })
}
