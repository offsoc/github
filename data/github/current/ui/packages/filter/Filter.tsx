import {type TestIdProps, testIdProps} from '@github-ui/test-id-props'
import {Box, type SxProp, useResponsiveValue} from '@primer/react'
import type {ResponsiveValue} from '@primer/react/lib-esm/hooks/useResponsiveValue'
import type React from 'react'
import {useCallback, useState} from 'react'

import {AdvancedFilterDialog, type FilterButtonVariant} from './advanced-filter-dialog/AdvancedFilterDialog'
import {ClearButton} from './clear-button/ClearButton'
import {defaultFilterSettings} from './constants/defaults'
import {FilterContextProvider} from './context/RootContext'
import {FilterQuery} from './filter-query'
import {FilterInputIcon} from './FilterInputIcon'
import {FilterInputWrapper} from './FilterInputWrapper'
import {Input} from './input/Input'
import {SubmitButton} from './input/SubmitButton'
import {SuggestionsList} from './suggestions/SuggestionsList'
import type {FilterProvider, FilterSettings, FilterVariant, SubmitEvent} from './types'
import {ValidationMessage} from './ValidationMessage'

// Exports from the root of the package
export * from './constants/defaults'
export * from './filter-query'
export * from './FilterRevert'
export * from './types'
export * from './ValidationMessage'

const filterInputWrapperStyles = ({variant, onSubmit}: {variant: FilterProps['variant']; onSubmit: boolean}) => {
  const leftRounded = variant === 'input'
  const rightRounded = (variant === 'input' && !onSubmit) || (variant !== 'input' && !onSubmit)

  return {
    display: 'grid',
    gridTemplateColumns: onSubmit ? '1fr min-content' : 'min-content 1fr min-content',
    alignItems: 'center',
    pl: onSubmit ? '12px' : 1,
    pr: '1px',
    width: '100%',
    borderStyle: 'solid',
    borderWidth: '1px',
    borderColor: 'border.default',
    marginInlineEnd: -1,
    ...(leftRounded && {
      borderTopLeftRadius: 2,
      borderBottomLeftRadius: 2,
    }),
    ...(rightRounded && {
      borderTopRightRadius: 2,
      borderBottomRightRadius: 2,
    }),
    ':has(input:focus-visible)': {
      zIndex: 1,
      outline: '2px solid var(--borderColor-accent-emphasis, var(--color-accent-emphasis))',
      outlineOffset: '-2px',
    },
  }
}

export type FilterProps = {
  /** ID of the Filter Component */
  id: string
  /** Change event callback, passing the updated filter value */
  onChange?: (value: string) => void
  /** Event callback, triggered by the filter query being parsed, and passing the updated filter query request*/
  onParse?: (request: FilterQuery) => void
  /** Submit event callback, triggered by Enter or pressing Submit, and passing the updated filter query and the
   * originating event type */
  onSubmit?: (request: FilterQuery, eventType: SubmitEvent) => void
  /** Array of FilterProviders that are being used/supported */
  providers: FilterProvider[]
  /** Context to be applied for all API queries */
  context?: Record<string, string>
  /** Variant of the filter button, whether to show the title or just the icon */
  filterButtonVariant?: FilterButtonVariant
  /** Initial value to use for the input. DO NOT USE in conjuction with `filterValue`.
   * Only use this if you want the Filter to have an initial value but remain uncontrolled. */
  initialFilterValue?: string
  /** Supplied filter value to use. By using this, the Filter is externally controlled
   * and will not use it's own state. */
  filterValue?: string
  /** React Ref to the input element */
  inputRef?: React.RefObject<HTMLInputElement>
  /** Label displayed atop the Filter bar. This is only shown when `visuallyHideLabel` is `false` */
  label: string
  /** Input keydown event callback */
  onKeyDown?: React.KeyboardEventHandler<Element>
  /** Validation callback, passing the validation message and the validation query */
  onValidation?: (messages: string[], filterQuery: FilterQuery) => void
  /** Placeholder text for the input */
  placeholder?: string
  /** Settings for the Filter, such as support of the legacy No filter provider and alias matching */
  settings?: FilterSettings
  /** Whether to show any validation messages or not. If this is disabled, use `onValidation` to receive the messages to display elsewhere */
  showValidationMessage?: boolean
  /** Variant of the Filter to render. `button` or `input` for one or the other, `full` (default) renders both.  */
  variant?: FilterVariant
  /** Whether to visually hide the Label or not */
  visuallyHideLabel?: boolean
} & TestIdProps &
  SxProp

const defaultFilterQuery = new FilterQuery()

export function Filter({
  context,
  'data-testid': dataTestId,
  filterButtonVariant = 'normal',
  filterValue,
  id,
  initialFilterValue,
  inputRef,
  label,
  onChange,
  onParse,
  onSubmit,
  onKeyDown,
  onValidation,
  placeholder,
  providers,
  settings,
  showValidationMessage = true,
  sx,
  variant = 'full',
  visuallyHideLabel = true,
}: FilterProps) {
  const [validationMessage, setValidationMessage] = useState<string[]>([])
  const hasValidationMessage = showValidationMessage && validationMessage.length > 0

  const validationCallback = useCallback(
    (messages: string[] = [], filterQuery: FilterQuery = defaultFilterQuery) => {
      setValidationMessage(messages)
      onValidation?.(messages, filterQuery)
    },
    [onValidation],
  )

  const filterButtonResponsive = useResponsiveValue<ResponsiveValue<FilterButtonVariant>, FilterButtonVariant>(
    {regular: filterButtonVariant, narrow: variant === 'button' ? 'normal' : 'compact'},
    filterButtonVariant,
  )

  const externallyControlled = typeof filterValue === 'string'
  const [innerValue, setInnerValue] = useState<string>(initialFilterValue || '')

  const changeCallback = useCallback(
    (value: string) => {
      if (!externallyControlled) {
        setInnerValue(value)
      }
      onChange?.(value)
    },
    [externallyControlled, onChange],
  )

  return (
    <FilterContextProvider
      context={context}
      inputRef={inputRef}
      rawFilter={externallyControlled ? filterValue : innerValue}
      onChange={changeCallback}
      onParse={onParse}
      onSubmit={onSubmit}
      onValidation={validationCallback}
      providers={providers}
      settings={{...defaultFilterSettings, ...settings}}
      variant={variant}
    >
      <Box
        id={id}
        role="form"
        className="FormControl FormControl--fullWidth"
        sx={{...sx, gap: 0}}
        {...testIdProps(dataTestId ?? 'filter')}
      >
        <label
          id={`${id}-label`}
          htmlFor={`${id}-input`}
          className={`FormControl-label ${visuallyHideLabel ? 'sr-only' : ''}`}
          {...testIdProps('filter-bar-label')}
        >
          {label}
        </label>
        <Box sx={{display: 'grid', gap: 3, width: '100%', position: 'relative'}}>
          <FilterInputWrapper isStandalone={variant === 'button'}>
            {variant !== 'input' && (
              <AdvancedFilterDialog isStandalone={variant === 'button'} filterButtonVariant={filterButtonResponsive} />
            )}
            {variant !== 'button' && (
              <>
                <Box
                  sx={filterInputWrapperStyles({
                    variant,
                    onSubmit: !!onSubmit,
                  })}
                >
                  {!onSubmit && <FilterInputIcon />}
                  <Input
                    id={id}
                    placeholder={placeholder}
                    onKeyDown={onKeyDown}
                    hasValidationMessage={hasValidationMessage}
                  />
                  <ClearButton />
                </Box>
                {onSubmit && <SubmitButton />}
              </>
            )}
          </FilterInputWrapper>
          <SuggestionsList id={id} />
          {/* This needs to move outside the Filter component because we can't predict where the consumer wants to put the error message. In memex this would go above the table view. */}
          {showValidationMessage && <ValidationMessage messages={validationMessage} id={`${id}-validation-message`} />}
        </Box>
      </Box>
    </FilterContextProvider>
  )
}
