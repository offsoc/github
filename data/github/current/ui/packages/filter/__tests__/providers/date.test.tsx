import {render} from '@github-ui/react-core/test-utils'
import {screen, waitFor} from '@testing-library/react'

import {TIME_RANGE_VALUES} from '../../constants/dynamic-filter-values'
import {Filter} from '../../Filter'
import {ClosedFilterProvider, CreatedFilterProvider, MergedFilterProvider, UpdatedFilterProvider} from '../../providers'
import {updateFilterValue} from '../../test-utils'
import {
  appendToFilterAndRenderAsyncSuggestions,
  expectDateSuggestionsToMatchSnapshot,
  expectFilterValueToBe,
  selectSuggestion,
  setupAsyncErrorHandler,
} from '../utils/helpers'

describe('Date', () => {
  setupAsyncErrorHandler()

  it('should render date suggestions', async () => {
    const filterProviders = [new CreatedFilterProvider()]

    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('created:')

    expectDateSuggestionsToMatchSnapshot()
  })
  it('should render dynamic value when selecting "Today"', async () => {
    const filterProviders = [new CreatedFilterProvider()]

    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('created:')

    expectDateSuggestionsToMatchSnapshot()

    await selectSuggestion('Today')

    expectFilterValueToBe('created:@today')
  })

  it('should render dynamic value when selecting "Yesterday"', async () => {
    const filterProviders = [new CreatedFilterProvider()]

    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)
    await updateFilterValue('created:')

    expectDateSuggestionsToMatchSnapshot()

    await selectSuggestion('Yesterday')

    expectFilterValueToBe('created:@today-1d')
  })

  it('should render dynamic value when selecting "Past 7 days"', async () => {
    const filterProviders = [new CreatedFilterProvider()]

    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('created:')

    expectDateSuggestionsToMatchSnapshot()

    await selectSuggestion('Past 7 days')

    expectFilterValueToBe('created:>@today-1w')
  })

  it('should render dynamic value when selecting "Past 30 days"', async () => {
    const filterProviders = [new CreatedFilterProvider()]

    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('created:')

    expectDateSuggestionsToMatchSnapshot()

    await selectSuggestion('Past 30 days')

    const input = screen.getByRole('combobox')
    expect(input).toHaveValue('created:>@today-30d')
  })

  it('should render dynamic value when selecting "Past year"', async () => {
    const filterProviders = [new CreatedFilterProvider()]

    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('created:')

    expectDateSuggestionsToMatchSnapshot()

    await selectSuggestion('Past year')

    expectFilterValueToBe('created:>@today-1y')
  })

  it('should render static value when selecting current day', async () => {
    const filterProviders = [new CreatedFilterProvider()]

    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('created:')

    expectDateSuggestionsToMatchSnapshot()

    const value = typeof TIME_RANGE_VALUES[5]?.value === 'function' ? TIME_RANGE_VALUES[5]?.value() : ''
    await selectSuggestion(TIME_RANGE_VALUES[5]?.displayName ?? '')
    expectFilterValueToBe(`created:${value}`)
  })

  it('should filter suggestions based on "before" operator and still retain operator on selection', async () => {
    const filterProviders = [new CreatedFilterProvider()]

    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('created:<')

    expectDateSuggestionsToMatchSnapshot()

    await selectSuggestion('Yesterday')

    expectFilterValueToBe('created:<@today-1d')
  })

  it('should not filter suggestions based on "after" operator', async () => {
    const filterProviders = [new CreatedFilterProvider()]

    const {user} = render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('created:>')

    expectDateSuggestionsToMatchSnapshot()

    await user.keyboard('{ArrowDown>4}')
    await user.keyboard('{Enter}')

    expectFilterValueToBe('created:>@today-30d')
  })

  it('should support between operator', async () => {
    const filterProviders = [new CreatedFilterProvider()]

    render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

    await updateFilterValue('created:')

    await selectSuggestion('Yesterday')

    await appendToFilterAndRenderAsyncSuggestions('..@today')

    expectFilterValueToBe('created:@today-1d..@today')
  })

  describe('Created', () => {
    it('should filter and select created suggestions based on name', async () => {
      const filterProviders = [new CreatedFilterProvider()]
      render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

      await updateFilterValue('created:@tod')

      expectDateSuggestionsToMatchSnapshot()

      await selectSuggestion('Today')

      expectFilterValueToBe('created:@today')
    })

    it('should not show any created suggestions based on invalid name', async () => {
      const filterProviders = [new CreatedFilterProvider()]
      render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

      await updateFilterValue('tttrrr')

      await waitFor(() => {
        expect(screen.getByTestId('filter-results')).toBeEmptyDOMElement()
      })
    })
  })

  describe('Closed', () => {
    it('should filter and select closed suggestions based on name', async () => {
      const filterProviders = [new ClosedFilterProvider()]
      render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

      await updateFilterValue('closed:@tod')

      expectDateSuggestionsToMatchSnapshot()

      await selectSuggestion('Today')

      expectFilterValueToBe('closed:@today')
    })

    it('should not show any closed suggestions based on invalid name', async () => {
      const filterProviders = [new ClosedFilterProvider()]
      render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

      await updateFilterValue('tttrrr')

      await waitFor(() => {
        expect(screen.getByTestId('filter-results')).toBeEmptyDOMElement()
      })
    })
  })

  describe('Merged', () => {
    it('should filter and select suggestions based on name', async () => {
      const filterProviders = [new MergedFilterProvider()]
      render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

      await updateFilterValue('merged:@tod')

      expectDateSuggestionsToMatchSnapshot()

      await selectSuggestion('Today')

      expectFilterValueToBe('merged:@today')
    })

    it('should not show any suggestions based on invalid name', async () => {
      const filterProviders = [new MergedFilterProvider()]
      render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

      await updateFilterValue('tttrrr')

      await waitFor(() => {
        expect(screen.getByTestId('filter-results')).toBeEmptyDOMElement()
      })
    })
  })

  describe('Updated', () => {
    it('should filter and select suggestions based on name', async () => {
      const filterProviders = [new UpdatedFilterProvider()]
      render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

      await updateFilterValue('updated:@tod')

      expectDateSuggestionsToMatchSnapshot()

      await selectSuggestion('Today')

      expectFilterValueToBe('updated:@today')
    })

    it('should not show any suggestions based on invalid name', async () => {
      const filterProviders = [new UpdatedFilterProvider()]
      render(<Filter id="test-filter-bar" label="Filter" providers={filterProviders} />)

      await updateFilterValue('tttrrr')

      await waitFor(() => {
        expect(screen.getByTestId('filter-results')).toBeEmptyDOMElement()
      })
    })
  })
})
