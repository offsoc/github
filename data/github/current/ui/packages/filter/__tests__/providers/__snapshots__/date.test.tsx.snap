// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Closed should filter and select closed suggestions based on name 1`] = `
[
  "Today",
  "Yesterday",
  "Past 7 days",
  "Past 30 days",
  "Past year",
]
`;

exports[`Date Created should filter and select created suggestions based on name 1`] = `
[
  "Today",
  "Yesterday",
  "Past 7 days",
  "Past 30 days",
  "Past year",
]
`;

exports[`Date Merged should filter and select suggestions based on name 1`] = `
[
  "Today",
  "Yesterday",
  "Past 7 days",
  "Past 30 days",
  "Past year",
]
`;

exports[`Date Updated should filter and select suggestions based on name 1`] = `
[
  "Today",
  "Yesterday",
  "Past 7 days",
  "Past 30 days",
  "Past year",
]
`;

exports[`Date should filter suggestions based on "before" operator and still retain operator on selection 1`] = `
[
  "Today",
  "Yesterday",
]
`;

exports[`Date should not filter suggestions based on "after" operator 1`] = `
[
  "Today",
  "Yesterday",
  "Past 7 days",
  "Past 30 days",
  "Past year",
]
`;

exports[`Date should render date suggestions 1`] = `
[
  "Today",
  "Yesterday",
  "Past 7 days",
  "Past 30 days",
  "Past year",
]
`;

exports[`Date should render dynamic value when selecting "Past 7 days" 1`] = `
[
  "Today",
  "Yesterday",
  "Past 7 days",
  "Past 30 days",
  "Past year",
]
`;

exports[`Date should render dynamic value when selecting "Past 30 days" 1`] = `
[
  "Today",
  "Yesterday",
  "Past 7 days",
  "Past 30 days",
  "Past year",
]
`;

exports[`Date should render dynamic value when selecting "Past year" 1`] = `
[
  "Today",
  "Yesterday",
  "Past 7 days",
  "Past 30 days",
  "Past year",
]
`;

exports[`Date should render dynamic value when selecting "Today" 1`] = `
[
  "Today",
  "Yesterday",
  "Past 7 days",
  "Past 30 days",
  "Past year",
]
`;

exports[`Date should render dynamic value when selecting "Yesterday" 1`] = `
[
  "Today",
  "Yesterday",
  "Past 7 days",
  "Past 30 days",
  "Past year",
]
`;

exports[`Date should render static value when selecting current day 1`] = `
[
  "Today",
  "Yesterday",
  "Past 7 days",
  "Past 30 days",
  "Past year",
]
`;
